redis-persistence/src/main/java/com/netflix/conductor/jedis/DynomiteJedisProvider.java

```diff
@@ -51,11 +67,23 @@ public JedisCommands get() {
                         configuration.getMaxConnectionsPerHost()
                 );
 
-        return new DynoJedisClient.Builder()
+        final DynoJedisClient.Builder builder = new DynoJedisClient.Builder()
                 .withHostSupplier(hostSupplier)
                 .withApplicationName(configuration.getAppId())
                 .withDynomiteClusterName(configuration.getClusterName())
-                .withCPConfig(connectionPoolConfiguration)
-                .build();
+                .withCPConfig(connectionPoolConfiguration);
+
+        if (this.configuration.shouldUseSSL()) {
+                try {
+                        final SSLContext sslContext = createAndInitSSLContext();
+                        builder.withSSLSocketFactory(sslContext.getSocketFactory());
```

Comment:
To make it extensible, please inject `SSLSocketFactory` through a Provider here.